sql/core/src/main/scala/org/apache/spark/sql/execution/ExistingRDD.scala

```diff
@@ -319,3 +319,43 @@ case class RDDScanExec(
 
   override def getStream: Option[SparkDataStream] = stream
 }
+
+/**
+ * A physical plan node for `OneRowRelation` for scans with no 'FROM' clause.
+ *
+ * We do not extend `RDDScanExec` in order to avoid complexity due to `TreeNode.makeCopy` and
+ * `TreeNode`'s general use of reflection.
+ */
+case class OneRowRelationExec() extends LeafExecNode
+  with InputRDDCodegen {
+
+  override val nodeName: String = s"Scan OneRowRelation"
+
+  override val output: Seq[Attribute] = Nil
+
+  private val rdd: RDD[InternalRow] = session.sparkContext.parallelize(Seq(InternalRow.empty), 1)
```

Comment:
thinking about it more, I think we can avoid serializing any row:
```
session.sparkContext.parallelize(Nil, 1).mapPartitionsInternal { _ =>
  val proj = UnsafeProjection.create(Nil)
  Iterator(proj.apply(InternalRow.empty))
}
```

Reply:
done, thanks for the help! ended up doing 
```
private val rdd: RDD[InternalRow] = {
    val numOutputRows = longMetric("numOutputRows")
    session
      .sparkContext
      .parallelize(Seq(InternalRow()), 1)
      .mapPartitionsInternal { _ =>
        val proj = UnsafeProjection.create(Seq.empty[Expression])
        Iterator(proj.apply(InternalRow.empty)).map { r =>
          numOutputRows += 1
          r
        }
      }
  }
```
to ensure the metrics are filled properly.